### PR TITLE
External Initiator for Klaytn

### PR DESCRIPTION
**Pull request by winnie-byun**
External Initiator for Klaytn.

`subscribe` API in Klaytn works similar with eth's. It is different from eth that klaytn uses 'klay' instead of 'eth' in method.
If you are subscribing something other than Log, it could have different struct from eth's. (e.g. Structs such as Headers are different.)

I've tested Klaytn EI with chainlink node and  attaching to Klaytn testnet (Baobab)
: https://github.com/winnie-byun/demo-klaytn-chainlink

### CODE REVIEW

**Comment by aalu1418** [comment]
@winnie-byun Hello! I've been assigned to review the Klatyn EI implementation. After a first look, here's some feedback so we can get your PR merged!

* It seems like you based the `klatyn.go` file from the `eth.go` file which is a good start. There are some functions in creating the filter query and parsing the event logs that would be needed to make it work with our on-chain smart contracts (example from the BSC initiator).
  * Creating the filter query: https://github.com/smartcontractkit/external-initiator/blob/master/blockchain/binance-smart-chain.go#L27
  * Parsing: https://github.com/smartcontractkit/external-initiator/blob/master/blockchain/binance-smart-chain.go#L133
* Make sure all the integration tests run successfully. They can be run with instructions in the root readme
* There also seems to be some linting errors, these can usually be solved with `go fmt`
* Also, please rebase so the latest changes are included

Thanks!

**Comment by winnie-byun** [comment]
@aalu1418 I changed the code like the BSC initiator and rebased.
The code is already `go fmt`ed.
I tested that with a chainlink core and it runs with no error. I will update after I tested with klaytn chainlink adapter.
Please take a look.

File: integration/mock-client/blockchain/klaytn_test.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+			if (err != nil) != tt.wantErr {
+				t.Errorf("handleKlaytnSubscribe() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("handleKlaytnSubscribe() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (aalu1418) [inline integration/mock-client/blockchain/klaytn_test.go:192]: Please add testing functions for `klaytnLogRequestToResponse` and `handleKlaytnGetLogs` 😄

File: blockchain/klaytn.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+			}
+		}
+
+	default:
+		logger.Errorw(ErrSubscriberType.Error(), "type", k.p)
+		return nil, false
+	}
+
+	return events, true
+}
```
Comment (aalu1418) [inline blockchain/klaytn.go:204]: Seems like you're missing the `blockchain/klaytn_test.go` file for testing the various functions?

**Review by aalu1418 — CHANGES_REQUESTED** [review]
Looking pretty good!
Most of the feedback is around some missing pieces that I noticed.

**Comment by aalu1418** [comment]
Requested changes will be addressed in a separate PR

**Pull request merged**